modules/acm_sku/acm_sku.install

```diff
@@ -109,43 +113,22 @@ function acm_sku_update_8001() {
  *   "Null value not allowed"
  */
 function acm_sku_update_8101() {
-  try {
-    $status_storage_schema = [
-      'acm_sku' => [
-        'fields' => [
-          'status' => [
-            'type' => 'int',
-            'size' => 'tiny',
-            'not null' => TRUE,
-          ],
-        ],
-      ],
-    ];
-    \Drupal::keyValue('entity.storage_schema.sql')
-      ->set('acm_sku.field_schema_data.status', $status_storage_schema);
 
-    $database = \Drupal::database();
-    $table_name = 'acm_sku';
-    $column_name = 'status';
-    if ($database->schema()->tableExists($table_name) && !$database->schema()->fieldExists($table_name, $column_name)) {
-      $database->schema()->addField($table_name, $column_name, [
-        'type' => 'int',
-        'size' => 'tiny',
-        'not null' => TRUE,
-        'description' => 'Whether the SKU is available or not.',
-        'default' => 1,
-      ]);
-
-      \Drupal::entityDefinitionUpdateManager()->applyUpdates();
-    }
-  }
-  catch (\Exception $e) {
-    \Drupal::logger('acm_sku')->error($e->getMessage());
-  }
+  $storage_definition = BaseFieldDefinition::create('boolean')
+    ->setLabel(t('Status'))
+    ->setDescription(t('Whether the SKU is available or not.'))
+    ->setDefaultValue(TRUE);
+
+  \Drupal::entityDefinitionUpdateManager()
+    ->installFieldStorageDefinition('status', 'acm_sku', 'acm_sku', $storage_definition);
```

Comment:
Do these functions throw exceptions? If yes, please catch and log as before. If no, do they write to the log?

Reply:
I don't think so, this is using the same pattern that Node uses in core, see  the following:
https://git.drupalcode.org/project/drupal/-/blob/8.8.x/core/modules/node/node.install#L169